src/ice/agent.rs

```diff
@@ -559,25 +535,7 @@ impl IceAgent {
             .filter(|v| v.addr().is_ipv6() == ip.is_ipv6())
             .count() as u32;
 
-        // For relayed candidates, we add a "punishment" to the local preference
-        // if the base address differs in the IP version from the allocated address
-        // of the candidate.
-        // This punishment ensures that we prefer relayed within the same IP version,
-        // e.g. IPv4 <> IPv4 over ones that translate between IP version, e.g. IPv4 <> IPv6.
-        let relay_across_ip_version_punishment = if c.kind() == CandidateKind::Relayed {
-            if c.base().is_ipv4() != ip.is_ipv4() {
-                1000
-            } else {
-                0
-            }
-        } else {
-            0
-        };
-
-        let pref = counter_start - same_kind * 2 - relay_across_ip_version_punishment;
-        trace!("Calculated local preference: {}", pref);
-
-        c.set_local_preference(pref);
+        c.adjust_local_preference(same_kind * 2);
```

Comment:
It is a bit smelly that this code knows that it needs to adjust by `same_kind * 2`. I contemplated about narrowing the API even further to just say `adjust_local_preference_for_nth_candidate` but I thought that was a bit too specific. Open to other ideas.